Implement get_env_plugin_dirs() to return a list of directory names
specified by environment variable NSPLUGIN_DIRS.
Adjust get_mozilla_plugin_dirs() to append the mentioned list at the end.
Adjust npconfig_CFLAGS in Makefile to include glib during compile.

 -- Sasa Bodiroza <jazzva@gmail.com>  Wed, 01 Apr 2009 21:50:00 +0200
Index: ubuntu.1.2.2-0ubuntu3/src/npw-config.c
===================================================================
--- ubuntu.1.2.2-0ubuntu3.orig/src/npw-config.c	2009-04-02 16:29:55.000000000 +0200
+++ ubuntu.1.2.2-0ubuntu3/src/npw-config.c	2009-04-02 16:32:56.000000000 +0200
@@ -38,6 +38,7 @@
 #include <pwd.h>
 #include <dirent.h>
 
+#include <glib.h>
 
 static bool g_auto = false;
 static bool g_verbose = false;
@@ -213,6 +214,21 @@
   return plugin_path;
 }
 
+const gchar **get_env_plugin_dirs(int *count)
+{
+  char *ns_plugin_dir = getenv("NSPLUGIN_DIRS");
+  if (ns_plugin_dir == NULL)
+    return NULL;
+  *count = 0;
+  const gchar **ns_plugin_dirs = g_strsplit((gchar *)ns_plugin_dir, ":", -1);
+  const gchar **iter = ns_plugin_dirs;
+  while (*iter) {
+    (*count)++;
+    iter++;
+  }
+  return ns_plugin_dirs;
+}
+
 static const char **get_mozilla_plugin_dirs(void)
 {
   static const char *default_dirs[] = {
@@ -257,13 +273,23 @@
   };
 
   const int n_default_dirs = (sizeof(default_dirs) / sizeof(default_dirs[0]));
-  const char **dirs = malloc((n_default_dirs + 2) * sizeof(dirs[0]));
+  int n_env_dirs;
+  const gchar **env_dirs = get_env_plugin_dirs(&n_env_dirs);
+  const char **dirs = malloc((n_default_dirs + n_env_dirs + 2) * sizeof(dirs[0]));
   int i, j;
   for (i = 0, j = 0; i < n_default_dirs; i++) {
 	const char *dir = default_dirs[i];
 	if (dir && access(dir, F_OK) == 0)
 	  dirs[j++] = dir;
   }
+  if (env_dirs) {
+    const gchar **iter = env_dirs;
+    while (*iter) {
+	  if (*iter && access(*iter, F_OK) == 0)
+	    dirs[j++] = (char *)*iter;
+        iter++;
+    }
+  }
   dirs[j++] = get_user_mozilla_plugin_dir();
   dirs[j] = NULL;
   return dirs;
Index: ubuntu.1.2.2-0ubuntu3/Makefile
===================================================================
--- ubuntu.1.2.2-0ubuntu3.orig/Makefile	2009-04-02 16:29:55.000000000 +0200
+++ ubuntu.1.2.2-0ubuntu3/Makefile	2009-04-02 16:29:55.000000000 +0200
@@ -174,7 +174,8 @@
 npconfig_RAWSRCS = npw-config.c
 npconfig_SOURCES = $(npconfig_RAWSRCS:%.c=$(SRC_PATH)/src/%.c)
 npconfig_OBJECTS = $(npconfig_RAWSRCS:%.c=npconfig-%.o)
-npconfig_LDFLAGS = $(libdl_LDFLAGS)
+npconfig_CFLAGS  = $(GLIB_CFLAGS)
+npconfig_LDFLAGS = $(GLIB_LDFLAGS) $(libdl_LDFLAGS)
 ifneq (,$(findstring $(OS),netbsd dragonfly))
 # We will try to dlopen() the native plugin library. If that lib is
 # linked against libpthread, then so must our program too.
@@ -410,7 +411,7 @@
 	$(CC) -o $@ $(npconfig_OBJECTS) $(npconfig_LDFLAGS)
 
 npconfig-%.o: $(SRC_PATH)/src/%.c
-	$(CC) -o $@ -c $< $(CPPFLAGS) $(CFLAGS)
+	$(CC) -o $@ -c $< $(CPPFLAGS) $(CFLAGS) $(npconfig_CFLAGS)
 
 $(nploader_PROGRAM): $(nploader_SOURCES)
 	sed -e "s|%NPW_LIBDIR%|$(pkglibdir)|" $< > $@
